### PR TITLE
Istiod templates

### PR DESCRIPTION
**Pull request by costinm**
Initial templates - the intent is to share the first step ( cluster install ) so both istio and istiod can run 
at the same time, for transition.

### CODE REVIEW

File: kustomize/istiod/istiod.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      labels:
+        app: istiod
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: istiod
+      #      priorityClassName: "{{ .Values.global.priorityClassName }}"
+      containers:
+        - name: istiod
+          image: gcr.io/costin-istio/istiod:latest
```
Comment (howardjohn) [inline kustomize/istiod/istiod.yaml]: TODO replace with official istio ones once we build them

File: kustomize/istiod/istiod.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          #command: ["sleep", "50000" ]
+          ports:
+            - containerPort: 8080
+            - containerPort: 15010
+          readinessProbe:
+            httpGet:
+              path: /ready
+              port: 8080
+            initialDelaySeconds: 5
+            periodSeconds: 30
```
Comment (howardjohn) [inline kustomize/istiod/istiod.yaml]:
```suggestion
            periodSeconds: 5
````

to match recent changes to pilot

**Review by howardjohn — APPROVED** [review]

File: kustomize/cluster/istiod.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    resources: ["*"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["networking.istio.io"]
+    resources: ["*"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["authentication.istio.io"]
+    resources: ["*"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["rbac.istio.io"]
+    resources: ["*"]
```
Comment (ayj) [inline kustomize/cluster/istiod.yaml]: You can replace this and lines 54-65 with the following.

```yaml
- apiGroups: ["config.istio.io", "networking.istio.io", "authentication.istio.io", "rbac.istio.io" ]
  resources: ["*"]
  verbs: ["get", "list", "watch"]
```

File: kustomize/istiod/istiod.yaml (hunk 1 above)
Comment (costinm) [reply kustomize/istiod/istiod.yaml]: Automerger too fast...

This one is obvious :-)

**Pull request merged**